File: bug12_missing_else_case/mainn.js
// Bug: missing else case — tag undefined for values not divisible by 3 or 5
function classify(v) {
    if (v % 3 === 0) return 0xAABBn;
    else if (v % 5 === 0) return 0xCCDDn;
    else return 0xEEEEn; // FIX: default value instead of undefined
}

function run() {
    let acc = 0x0F0F0F0F0F0F0F0Fn;

    for (let n = 1; n <= 10; n++) {
        let t = classify(n);

        // FIX: removed undefined check since classify always returns a value
        // if (t === undefined) {
        //     console.error("UnboundLocalError: local variable 'tag' referenced before assignment");
        //     process.exit(1);
        // }

        acc = BigInt.asUintN(64, (acc ^ (t * BigInt(n))) * 0x9E3779B97F4A7C15n);
        acc = BigInt.asUintN(64, (acc << 9n) | (acc >> 55n));
    }

    let buf = Buffer.alloc(8);
    buf.writeBigUInt64BE(acc, 0);
    console.log(buf.toString('base64'));
}

run();

File: bug12_missing_else_case/mainn.java
import java.util.Base64;
import java.nio.ByteBuffer;

public class Main {

    /* FIX: added default return so no sentinel/uninitialized case */
    static long classify(int v) {
        if (v % 3 == 0) return 0xAABBL;
        else if (v % 5 == 0) return 0xCCDDL;
        else return 0xEEEEL; // FIX: default value instead of -1
    }

    static void run() {
        long acc = 0x0F0F0F0F0F0F0F0FL;

        for (int n = 1; n <= 10; n++) {
            long t = classify(n);

            // FIX: removed sentinel check since classify always returns valid value
            // if (t == -1L) {
            //     System.err.println("UnboundLocalError: local variable 'tag' referenced before assignment");
            //     System.exit(1);
            // }

            acc = (acc ^ (t * (long)n)) * 0x9E3779B97F4A7C15L;
            acc &= 0xFFFFFFFFFFFFFFFFL;
            acc = ((acc << 9) | (acc >>> 55)) & 0xFFFFFFFFFFFFFFFFL;
        }

        ByteBuffer buf = ByteBuffer.allocate(8);
        buf.putLong(acc);
        System.out.println(Base64.getEncoder().encodeToString(buf.array()));
    }

    public static void main(String[] args) {
        run();
    }
}